addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1781,31 +1809,54 @@ const ListUserValueWidget = UserValueWidget.extend({
         if (this.addItemButton) {
             this.addItemButton.remove();
         }
-        if (this.isCustom) {
-            this.addItemButton = document.createElement('we-button');
-            this.addItemButton.textContent = this.addItemTitle;
-            this.addItemButton.classList.add('o_we_list_add_optional');
-            currentValues.forEach(el => this._addItemToTable(el, el));
+
+        if (this.createWidget) {
+            const selectedIds = currentValues.map(({ id }) => id)
+                .filter(id => typeof id === 'number');
+            this.createWidget.options.domain = ['!', ['id', 'in', selectedIds]];
+            this.createWidget.setValue('');
+            this.createWidget.inputEl.value = '';
+            $(this.createWidget.inputEl).trigger('input');
```

Comment:
What's the idea behind this line?

Reply:
Triggers the widget's search so that the records are up to date with the new domain and the empty input that might be leftover from a previous search. I used to call _search directly but since it's a private method I was hesitant about that.